packages/core/src/validate.js

```diff
@@ -275,3 +276,27 @@ export function isValid(schema, data) {
     return false;
   }
 }
+
+/**
+ * Validates data against a schema, returning true if the data is valid, or
+ * false otherwise. If the schema is invalid, then this function will return
+ * false.
+ */
+export function isValidAddDefaults(schema, data) {
+  try {
+    if ("properties" in schema) {
+      // This is a bit of a hack, ajv seems to ignore undefined values in validate, this needs to
+      // be looked into further, but setting default and option turned on to populate defaults
+      for (var key in schema.properties) {
+        // kind of fixes this validation for now
+        var property = schema.properties[key];
+        if (!("default" in property)) {
+          property["default"] = "";
+        }
+      }
+    }
+    return ajv.validate(schema, data);
+  } catch (e) {
+    return false;
+  }
+}
```

Comment:
Looking at this, further I think that AJV could be correct here, 

https://github.com/epoberezkin/ajv/issues/913

this could be confusing but is basically a case of making the field required in schema,

I am looking at adding const support to this also as const is only partually supported, it should basically have default value and be disabled.
 

https://github.com/epoberezkin/ajv/issues/913

this